src/NzbDrone.Core/Download/Clients/Porla/Porla.cs

```diff
@@ -0,0 +1,239 @@
+
+                // do we trust porla?
+                // if (torrent.download_rate > 0 && torrent.total_done > 0) { eta = TimeSpan.FromSeconds(torrent.total / (double)torrent.download_rate); }
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this, false),
+                    DownloadId = torrent.InfoHash.Hash,
+                    OutputPath = outputPath + torrent.Name,
+                    RemainingSize = torrent.Total,
+                    RemainingTime = eta,
+                    Title = torrent.Name,
+                    TotalSize = torrent.Size,
+                    SeedRatio = torrent.Ratio
+                };
+
+                if (!string.IsNullOrEmpty(torrent.Error))
+                {
+                    item.Status = DownloadItemStatus.Warning;
+                    item.Message = torrent.Error;
+                }
+
+                _logger.Debug($"Torrent {torrent.InfoHash.Hash} state is {torrent.State}");
+
+                // deal with moving_storage???
+                if (torrent.FinishedDuration > 0)
+                {
+                    item.Status = DownloadItemStatus.Completed;
+                }
+
+                // else if (torrent.state == HadoukenTorrentState.QueuedForChecking)
+                // {
+                //    item.Status = DownloadItemStatus.Queued;
+                // }
+                // else if (torrent.State == HadoukenTorrentState.Paused)
+                // {
+                //    item.Status = DownloadItemStatus.Paused;
+                // }
+                else
+                {
+                    item.Status = DownloadItemStatus.Downloading;
+                }
+
+                // torrent.state exists, can't quite tell if it's passthrough of https://libtorrent.org/reference-Torrent_Status.html#state_t
+
+                item.CanMoveFiles = item.CanBeRemoved = true; // usure of the restrictions here
+
+                items.Add(item);
+            }
+
+            return items;
+        }
+
+        public override void RemoveItem(DownloadClientItem item, bool deleteData)
+        {
+            // Kinda sucks we don't have a `RemoveItems`, porla has a batch interface for removals
+            PorlaTorrent[] singleItem = { new PorlaTorrent(item.DownloadId.ToLower(), "") };
+            _proxy.RemoveTorrent(Settings, deleteData, singleItem);
+
+            // when do we set item.Removed ?
+        }
+
+        public override DownloadClientInfo GetStatus()
+        {
+            // var preset = _proxy.ListPresets(Settings);
+            // var sessettings = _proxy.GetSessionSettings(Settings);
+
+            var status = new DownloadClientInfo
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                RemovesCompletedDownloads = false, // should be determined through Flags From  somewhere? I think session?
```

Comment:
This is based on settings configured in Porla. For example qbit, a user can configure qbit to either pause or remove completed downloads, if it's pause that will work with Sonarr, if it's remove it can be removed before Sonarr can import and we want to warn the user.

Reply:
Ok.
I'll dig through the docs again. I think you can set up that behavior. But IIRC, I think you can set that behavior on a per-trorrent bassis

just double checking: this code needs to be able to tell what is set up by checking in with porla correct?